src/components/Calendar/index.js

```diff
@@ -122,16 +317,26 @@ class CalendarComponent extends Component {
                         onChange={this.handleYearChange}
                     />
                 </StyledControlsContainer>
-                <table role="grid" aria-labelledby={this.monthLabelId}>
+                <StyledTable
+                    role="grid"
+                    ref={this.tableRef}
+                    aria-labelledby={this.monthLabelId}
+                    tabIndex="0"
```

Comment:
The approach followed by the links showed in the issue don't do this, instead they set tabIndex="0" only to the day that will get the focus and all other are tabIndex="-1", then when the use the keyboard to move focus to another day then the new focused day gets tabIndex="0"  and previous get tabIndex="-1". This should solve the problem with the onClick. I think it is better to follow this approach because it is not accessible to make the table focusable since what is focusable is the day to select.